core/src/main/java/org/jdbi/v3/core/generic/GenericTypes.java

```diff
@@ -147,13 +155,22 @@ public static Type resolveMapEntryType(Type mapType) {
      * @return the map entry type
      */
     public static Type resolveMapEntryType(Type keyType, Type valueType) {
-        return resolveMapEntryType(TypeToken.of(keyType), TypeToken.of(valueType));
+        return TypeFactory.parameterizedClass(Map.Entry.class, keyType, valueType);
     }
 
-    private static <K, V> Type resolveMapEntryType(TypeToken<K> keyType, TypeToken<V> valueType) {
-        return new TypeToken<Map.Entry<K, V>>() {}
-                .where(new TypeParameter<K>() {}, keyType)
-                .where(new TypeParameter<V>() {}, valueType)
-                .getType();
+    /**
+     * Creates a type of class {@code clazz} with {@code arguments} as type arguments.
+     * <p>
+     * For example: {@code parameterizedClass(Map.class, Integer.class, String.class)}
+     * returns the type {@code Map<Integer, String>}.
+     *
+     * @param clazz     Type class of the type to create
+     * @param arguments Type arguments for the variables of {@code clazz}, or null if these are not
+     *                  known.
+     * @return A {@link ParameterizedType}, or simply {@code clazz} if {@code arguments} is
+     * {@code null} or empty.
+     */
+    public static Type parameterizedClass(Class<?> clazz, Type... arguments) {
```

Comment:
Nice!

Could we rename this to `parameterizeClass` (without the `d`) to make the method name contain a verb?